app/lib/identity_tijuana/member_sync.rb

```diff
@@ -6,25 +6,55 @@ def self.export_member(member, sync_id)
       begin
         return if member.ghosting_started?
 
-        ext_id = MemberExternalId.find_by(system: 'tijuana', member: member)
-        user_id = ext_id.external_id if ext_id.present?
-        if user_id.present?
-          user = User.find_by(id: user_id)
-          ext_id.destroy! if user.blank?
+        ext_ids = MemberExternalId.where(system: 'tijuana', member: member).to_a
+        # Destroy any dangling external ID references, remove any
+        # external ids from the list if not found
+        ext_ids.select do |ext_id|
+          user = User.find_by(id: ext_id.external_id.to_i)
+          if user.blank?
+            Rails.logger.info(
+              "[IdentityTijuana::export_member]: " \
+              "Removing dangling external id for Id member #{member.id}, " \
+              "TJ user #{ext_id.external_id} " \
+              "(sync_id=#{sync_id}, sync_type=#{sync_type}, " \
+              "sync_direction=export_member)"
+            )
+            ext_id.destroy!
+          end
+          user.present?
         end
-        if user.blank?
-          sync_type = :merge
-          email = member.email
-          user = User.find_by(email: email) if email.present?
-          sync_type = :create if user.blank?
+        if ext_ids.count == 0
+          # XXX this will fail for de-normalised and/or cleaned emails
+          user = User.find_by(email: member.email) if member.email.present?
+
+          if user.nil?
```

Comment:
Shall we use `blank?` or `nil?` for consistency

Reply:
This is honestly one of my least favourite things about Rails and Ruby in general. Who the hell can remember [all this ridiculous minutiae](https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html)?

Since `user` should be a User model instance (not, say a string or a Boolean), I think I actually do want to use `nil?` here.